server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java

```diff
@@ -149,11 +149,14 @@ Response get() {
    Response getKeyMetadata() {
       CacheEntry<byte[], byte[]> ce = cache.getCacheEntry(key);
       if (ce != null) {
-         NumericVersion entryVersion = (NumericVersion) ce.getMetadata().version();
+         long version = 0;
+         EntryVersion entryVersion = ce.getMetadata().version();
+         if (entryVersion != null && entryVersion instanceof NumericVersion) {
```

Comment:
The Hot Rod protocol allows the version to travel in the wire as a `long` and `SimpleClusteredVersion` is a pair of longs, was not willing to change the protocol version to fix this CCE, wdyt, should it be handled in another jira? Behaviour is pretty much the same as before.